src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,163 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level; _⊔_)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties          as ModuleProperties
+import Function.Definitions
+import Relation.Binary.Reasoning.Setoid   as Reasoning
+import Algebra.Module.Morphism.Structures as MorphismStructures
+
+open import Axiom.DoubleNegationElimination
+open import Data.List
+open import Data.Product
+  using (Σ; _,_; ∃; Σ-syntax; ∃-syntax; _×_)
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+module A = Module modA
+open A using () renaming (Carrierᴹ to A)
+-- module A = ModuleProperties A.leftSemimodule
+-- open ModuleProperties A.leftSemimodule
+--   using () renaming
+--   ( non-zeroʳ to non-zeroʳᴬ
+--   ; non-zeroˡ to non-zeroˡᴬ
+--   )
+module B = Module modB
+open B using () renaming (Carrierᴹ to B)
+module B2 = ModuleProperties B.leftSemimodule
+-- open ModuleProperties B.leftSemimodule
+--   using () renaming
+--   ( non-zeroʳ to non-zeroʳᴮ
+--   ; non-zeroˡ to non-zeroˡᴮ
+--   )
+open CommutativeRing ring
+  using (_≈_; _*_; _+_; 0#) renaming (Carrier to S)
+
+module _
+  {f : A → B}
+  (isModuleHomomorphism : IsModuleHomomorphism f)
+  where
+
+  open IsModuleHomomorphism isModuleHomomorphism
+
+  -- f(x) ≈ 0 iff x ≈ 0, for linear non-trivial f
+  f𝟘≈𝟘 : f A.0ᴹ B.≈ᴹ B.0ᴹ
+  f𝟘≈𝟘 = begin
+    f A.0ᴹ           ≈⟨ ⟦⟧-cong (A.≈ᴹ-sym (A.*ₗ-zeroˡ A.0ᴹ)) ⟩
+    f (0# A.*ₗ A.0ᴹ) ≈⟨ *ₗ-homo 0# A.0ᴹ ⟩
+    0# B.*ₗ f A.0ᴹ   ≈⟨ B.*ₗ-zeroˡ (f A.0ᴹ) ⟩
+    B.0ᴹ             ∎
+    where open Reasoning B.≈ᴹ-setoid
+
+  x≈𝟘⇒fx≈𝟘 : ∀ {x} → x A.≈ᴹ A.0ᴹ → f x B.≈ᴹ B.0ᴹ
+  x≈𝟘⇒fx≈𝟘 {x} x≈𝟘 = begin
+    f x    ≈⟨ ⟦⟧-cong x≈𝟘 ⟩
+    f A.0ᴹ ≈⟨ f𝟘≈𝟘 ⟩
+    B.0ᴹ   ∎
+    where open Reasoning B.≈ᴹ-setoid
+
+  fx≉𝟘⇒x≉𝟘 : ∀ {x} → f x B.≉ᴹ B.0ᴹ → x A.≉ᴹ A.0ᴹ
+  fx≉𝟘⇒x≉𝟘 = contraposition x≈𝟘⇒fx≈𝟘
+
+  -- Zero is a unique output of linear map ≉ `const 𝟘`.
+  zero-unique : ∀ {x} →
```

Comment:
Might be useful to have a read of the [style guide ](https://github.com/agda/agda-stdlib/blob/master/notes/style-guide.md), especially the bit about indentation.